    /**
     * @test   ackermann throws an exception when an object is provided
     * @throws \Exception
     */

    public function testIncorrectTypeExceptionPrepareParameter()
    {
        // Given
        $m = new \stdClass();
        $n = $m;

        // Then
        $this->expectException(Exception\IncorrectTypeException::class);

        // When
        $int = ArbitraryInteger::ackermann($m, $n);
    }
